server/src/main/java/org/opensearch/action/search/SearchPhase.java

```diff
@@ -54,4 +55,34 @@ protected SearchPhase(String name) {
     public String getName() {
         return name;
     }
+
+    /**
+     * Returns the SearchPhase name as {@link SearchPhaseName}. Exception will come if SearchPhase name is not defined
+     * in {@link SearchPhaseName}
+     * @return {@link SearchPhaseName}
+     */
+    public SearchPhaseName getSearchPhaseName() {
+        return SearchPhaseName.valueOf(name.toUpperCase(Locale.ROOT));
+    }
+
+    /**
+     * Enum for different Search Phases in OpenSearch
+     * @opensearch.internal
+     */
+    public enum SearchPhaseName {
+        QUERY("query"),
+        FETCH("fetch"),
+        DFS_QUERY("dfs_query"),
+        EXPAND("expand");
```

Comment:
don't see `can_match` phase here. Can we please add that as well ?

Reply:
Sure will do that